src/EventBus.ts

```diff
@@ -89,16 +89,21 @@ export class EventBus {
   };
 
   subscribe = <T extends BusEvent>(
-    eventType: T["type"] | EventTypeDescriptor<T> | PredicateFn,
+    subscription: SubscriptionDef<T>,
     handler: (e: T) => void
   ) => {
-    if (isPredicateFn(eventType)) {
```

Comment:
We should support arrays of variable args here. I guess it will be a matter of making a function that does the check and test for an array to see if we run that function multiple times or just once.